packages/css/src/__tests__/generate-compression-map.test.ts

```diff
@@ -0,0 +1,139 @@
+    @media (min-width:1440px) and (max-width:1919px){._pbi4tcjq{display:block!important}}
+    @media (min-width:1920px) and (max-width:2559px){._16b9tcjq{display:block!important}}
+    @media (min-width:2560px) and (max-width:2999px){._jmaqtcjq{display:block!important}}
+    @media (min-width:3000px){._1q5htcjq{display:block!important}}
+    @media (min-width:800px) and (max-width:999px){._11x1tcjq{display:block!important}}
+    @media (min-width:800px){._121jagmp{display:none!important}}
+    @media (prefers-reduced-motion:reduce){._1bumglyw{animation:none}._sedtglyw{transition:none}}
+    @media screen and (-webkit-min-device-pixel-ratio:0){._14kw1hna >textarea{word-break:break-word}._mc2h1hna{word-break:break-word}}
+    @media screen and (-webkit-transition){._14fy1hna{word-break:break-word}._1vdp1hna >textarea{word-break:break-word}}
+    @media screen and (max-height:400px){._17gjpfqs{position:static}}
+    @media screen and (min-width:1300px){._1jhpoyl8{max-width:10vw}}
+`;
+
+  const baseResult = {
+    '154i14e6': 'a',
+    '14tk72c6': 'b',
+    '14n4stnw': 'c',
+    '13h81y44': 'd',
+    '1di6k6hx': 'e',
+    irr3k6hx: 'f',
+    jomrk6hx: 'g',
+    '10j7k6hx': 'h',
+    '1gg2glyw': 'i',
+    '1o3iglyw': 'j',
+    '1iohnqa1': 'k',
+    '5goinqa1': 'l',
+    jf4cnqa1: 'm',
+    xatrnqa1: 'n',
+    '1726nqa1': 'o',
+    '1iqunqa1': 'p',
+    '1ejunqa1': 'q',
+    '1lwpnqa1': 'r',
+    '6xf7nqa1': 's',
+    '1mb818uv': 't',
+    oga118uv: 'u',
+    '1n2onqa1': 'v',
+    '1k4fnqa1': 'w',
+    '1nrm18uv': 'x',
+    '1a3b18uv': 'y',
+    '9oik18uv': 'z',
+    '5bd618uv': '_',
+    '1ydc18uv': 'aa',
+    '1ohyglyw': 'ba',
+    '49pcglyw': 'ca',
+    ra3xglyw: 'da',
+    ksodglyw: 'ea',
+    q4asglyw: 'fa',
+    tpgfglyw: 'ga',
+    '1oxgru3m': 'ha',
+    '9h8h1e9r': 'ia',
+    f8pj1e9r: 'ja',
+    '30l31e9r': 'ka',
+    '10531e9r': 'la',
+    '1jhpoyl8': 'ma',
+    '11usglyw': 'na',
+    '11y7oza4': 'oa',
+    '11y7uu9g': 'pa',
+    l82t7vkz: 'qa',
+    j7o07vkz: 'ra',
+    '1od57vkz': 'sa',
+    l82tgktf: 'ta',
+    j7o0gktf: 'ua',
+    yksp1ssb: 'va',
+    '1b421ssb': 'wa',
+    s8ks18ws: 'xa',
+    u1wz1nty: 'ya',
+    hnu8tcjq: 'za',
+    jvpg11p5: '_a',
+    '1nwdwxkt': 'ab',
+    '1vlxckbl': 'bb',
+    kz8c16xz: 'cb',
+    '1jyu16xz': 'db',
+    '11et16xz': 'eb',
+    fgkv16xz: 'fb',
+    szna1wug: 'gb',
+    '13on1wug': 'hb',
+    '1f3k1wug': 'ib',
+    inid1wug: 'jb',
+    '12wp9ac1': 'kb',
+    jvpgglyw: 'lb',
+    pbi4tcjq: 'mb',
+    '16b9tcjq': 'nb',
+    jmaqtcjq: 'ob',
+    '1q5htcjq': 'pb',
+    '11x1tcjq': 'qb',
+    '121jagmp': 'rb',
+    '1bumglyw': 'sb',
+    sedtglyw: 'tb',
+    '14kw1hna': 'ub',
+    mc2h1hna: 'vb',
+    '14fy1hna': 'wb',
+    '1vdp1hna': 'xb',
+    '17gjpfqs': 'yb',
+  };
+  it('should generate class names without the old compression map', () => {
```

Comment:
Is the compression map supposed to be stable across builds? 
Here we are just testing that it gets generated; however, in the following test it seems that a different compressed class is generated for `17gjpfqs`

Reply:
I hope I understand the question correctly 🙂. Let me explain it.

> Is the compression map supposed to be stable across builds?

Yes, the compression map is supposed to be stable across builds. The way it achieves this is to take the compression map from the previous build as parameter:

```
generateCompressionMap = (
  css: string,
  oldClassNameCompressionMap?: { [index: string]: string }
):
```

For example, let's say we have the CSS:
```
._aaaabbbb { color: red }
._aaaacccc { color: black }
```
oldClassNameCompressionMap
```
{
   "aaaabbbb": "ddd"
}
```
`generateCompressionMap` returns new compression map
```
{
   "aaaacccc": "a"
   "aaaabbbb": "ddd" // Force "aaaabbbb" to be "ddd", which is same as oldClassNameCompressionMap
}
```

In the following test case, without providing an oldCompressionMap, '17gjpfqs' is compressed to "yb", but because we provide oldCompressionMap as  { "17gjpfqs": "a", 1vdp1hna': 'b',  '14fy1hna': 'c'}, '17gjpfqs' is compressed to 'a'. We want to ensure the new compression map keeps the same class name mapping from the oldCompressionMap.

Reply:
thanks for the explanation! Now I get it! I was looking at the  key/pair values that I didn't realise what the comparison was about  :)